doc/source/force-constants.rst

```diff
@@ -178,7 +190,45 @@ Some information must be provided as a pint ``Quantity`` with both a magnitude a
 For more details on the shape of each array, see the docstrings for each object.
 An example is shown below, assuming that the inputs are being loaded from existing ``.npy`` files:
 
-.. code-block:: py
+.. testsetup:: nacl_fc_convert_npy
+
+  import numpy as np
+  import json
+  from euphonic import ForceConstants
+
+  # Create ForceConstants constructor input files
+  fc_attrs = ['cell_vectors', 'atom_r', 'atom_type', 'atom_mass',
+              'force_constants', 'sc_matrix', 'cell_origins',
+              'born', 'dielectric']
+  fnames = [attr + '.npy' for attr in fc_attrs]
+  fc = ForceConstants.from_phonopy(path=get_phonopy_path('NaCl'),
+                                   summary_name='phonopy_nacl.yaml')
+  for attr, fname in zip(fc_attrs, fnames):
+      try:
+          val = getattr(fc, attr)
+      except AttributeError:
+          val = getattr(fc.crystal, attr)
+      try:
+          np.save(fname, val.magnitude)
+      except AttributeError:
+          np.save(fname, val)
+
+  convert_json = get_data_path(
+      'force_constants', 'NaCl_convert_full_fc_data.json')
+  with open(convert_json, 'r') as fp:
+      fc_convert_data = json.load(fp)
+  fnames_convert = ['force_constants_atomic_phase.npy',
+                    'supercell_atom_r.npy', 'unit_to_supercell_atom_idx.npy',
+                    'super_to_unit_cell_atom_idx.npy']
+  data_keys = ['force_constants', 'sc_atom_r', 'uc_to_sc_idx', 'sc_to_uc_idx']
+  for key, fname in zip(data_keys, fnames_convert):
+      np.save(fname, fc_convert_data[key])
+  fnames += fnames_convert
+  # Delete imports so any imports required by testcode must
+  # be correctly imported there
+  del np, ForceConstants
```

Comment:
I could add it, but the most important thing here is that `np`, `ForceConstants` are no longer direct references to the module/object (so that `testcode:: nacl_fc_convert_npy` fails if I forget to import them in the user example). If we don't delete the `sys.modules` reference, they won't get garbage collected but that shouldn't be much of an issue here I don't think...